### DISCUSSION TITLE
Pnp Search results webpart missing remove duplicates or dont remove duplicates

### DISCUSSION THREAD

**Discussion by krishnakishore1983** [microsoft-search/pnp-modern-search#2088] [Ideas]
Is it possible to configure below setting in pnp search results webpart

https://docs.microsoft.com/en-us/sharepoint/troubleshoot/search/near-duplicate-items-are-not-listed-in-search-results

I am unable to find remove or don't remove duplicates boolean flag in pnp as it's difficult and search count is mis match with oob search. Please guide me

**Comment by krishnakishore1983** [comment]
Reference link https://www.sharepointdiary.com/2019/07/sharepoint-online-fix-search-results-count-inconsistent-issue.html

**Reply by FranckyC** [reply]
Moved to discussion as it would be a new option to add.

**Reply by wobba** [reply]
I don't really want to expose this option as it causes more harm than good https://www.techmikael.com/2013/12/duplicate-trimming-in-sharepoint-2013.html. Supporting the collapse spec option would be better imo.

**Reply by krishnakishore1983** [reply]
It's already present in classic search results webpart why it's not same in pnp some business scenarios if we compare classic search results mismatch with pnp. May I know what harm even one of client asking to check dynamic 365 sharepoint search comparing with pnp there is some difference between these two search providers.

**Reply by wobba** [reply]
If you want to contribute a PR by adding the option you are free to do so, but we have no plans to do so from Franck or my side. Not everything from classic makes sense to bring forward - one of them being near duplicate detection which in my opinion was and still is a poor implementation. If you decide to PR, then do collapsespec with docs on using `DocumentSignature:1`.

**Reply by FranckyC** [reply]
The option has been added via PR #2135. It is still false but default but you can now control the behavior if desired.

**Reply by wobba** [reply]
@FranckyC I still don't think we should have added it, should rather have added the collapsespec feature. The oob feature is not a good one.

**Reply by Jayb127** [reply]
Sorry if this isn't the right place to ask this, but I'm wondering if anyone else is still experiencing issues with this despite the implementation of the trimming option. I've built a search configuration has a pretty narrow scope and limited test data, but I'm already seeing instances of duplicate removal. The results webpart indicates that there are 10 results, but only 8 appear in the webpart. 

Here are some other observations that may help: 
1. Occasionally, and seemingly randomly, 1 or more of the missing results will appear in the results webpart. This happens less than 25% of the time. Refreshing the page usually causes the result to disappear. 
2. Selecting filters will not make the missing result(s) appear, but the result count at the top of the results webpart will include that missing result. For example, after selecting an applicable filter, the count will say "4 results," but only 3 results will appear. 
3. Performing a keyword search will force the missing result to appear in the results every time. 

Are there any other options or workarounds to force these missing results to appear? Is there anything anyone can recommend that would allow me to verify that the problem here is SharePoint's interpretation of these items as duplicates as opposed to some other problem? 

Thank you.

**Reply by wobba** [reply]
As I've stated many times. The TrimDuplicate feature does not work for a lot of scenarios the way you think it would. CollapseSpec is a much better feature.

**Reply by Jayb127** [reply]
@wobba is there a way for me to incorporate this CollapseSpec into the configuration, or is this a change that would have to be made to the webparts?

**Reply by wobba** [reply]
It has to be coded in as a new setting for the web part.

**Reply by JoeAyre** [reply]
> The option has been added via PR #2135. It is still false but default but you can now control the behavior if desired.

Hi @wobba and @FranckyC - just reading this discussion and have a related question. @FranckyC - in your reply you state that it is still false by default. Does this mean that the trimduplicates is set to false? This is what I want - I agree with @wobba - the TrimDuplicates feature was a poor feature that caused more problems than it solved. However, if it is indeed false, why am I seeing a mismatch between the number of documents in a library and the number of search results returned when setting the search results web part query template to only look at that document library? It feels like duplicate trimming, hence this post. It's a frustrating issue for my implementation which I am keen to get to the bottom of, so hope you can confirm that the pnp-modern-search-parts-v4 version 4.5.4.0 has TrimDuplicates defaulted to false. Thanks, Joe

**Reply by wobba** [reply]
@JoeAyre check the search request using the debug panel (network tab) in your browser for what the 4.5.4 query looks like. Been so long I cannot remember :) You should be able to see if it set's the flag or not.

**Reply by JoeAyre** [reply]
Hi @wobba - thanks for this, although I'm unsure what I should be looking for in the Developer Toolbar (F12) Network tab. Is there something in particular I should be searching for? I've tried 'TrimDuplicates' but can't see anything. Thanks, Joe

**Reply by JoeAyre** [reply]
Hi again @wobba. Just to close this one off, looking at the documentation, default for TrimDuplicates is off (a good thing in my opinion). A toggle to turn it on was briefly introduced in v4.7.0. Regardless, it turned out a library reindex was required yo resolve my issue. The timer job for some reason must have got delayed in SharePoint Online itself. Happy to say though that the issue doesn't appear to be with the modern search solution. Thanks, Joe
